src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/SimpleTypeInfos.cs

```diff
@@ -288,13 +288,18 @@ public NullableTypeInfo(Type type, List<Type> recursionCheck)
             this.valueInfo.WriteMetadata(group, "Value", format);
         }
 
+#if !ES_BUILD_STANDALONE
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2072:UnrecognizedReflectionPattern",
+                Justification = "The underlying type of Nullable<T> must be defaultable")]
+#endif
         public override void WriteData(PropertyValue value)
         {
-            // It's not currently possible to get the HasValue property of a nullable type through reflection when the
-            // value is null. Instead, we simply check that the nullable is not null.
-            bool hasValue = value.ReferenceValue != null;
+            object? refVal = value.ReferenceValue;
+            bool hasValue = refVal is not null;
             TraceLoggingDataCollector.AddScalar(hasValue);
-            PropertyValue val = hasValue ? valueGetter(value) : valueInfo.PropertyValueFactory(Activator.CreateInstance(valueInfo.DataType));
+            PropertyValue val = valueInfo.PropertyValueFactory(hasValue
+                ? refVal
+                : RuntimeHelpers.GetUninitializedObject(valueInfo.DataType));
```

Comment:
Looks like you'll need to #if around this for `ES_BUILD_STANDALONE`, since `RuntimeHelpers.GetUninitializedObject` doesn't exist in that TargetFramework.

Reply:
(The API to call outside of that ifdef is [`FormatterServices.GetUninitializedObject`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.formatterservices.getuninitializedobject?view=net-5.0))